Bugsnag/Delivery/BSGEventUploader.m

```diff
@@ -82,7 +92,68 @@ - (void)uploadEvent:(BugsnagEvent *)event completionHandler:(nullable void (^)(v
     [self.uploadQueue addOperation:operation];
 }
 
++ (void)synchronouslyUploadAtomicReportsWithConfiguration:(BugsnagConfiguration *)configuration {
+
+    BugsnagNotifier *notifier = [BugsnagNotifier new];
+    NSFileManager *fm = NSFileManager.defaultManager;
+    NSError *error = nil;
+
+    NSString *container = [BSGFileLocations atomicDirectoryContainer];
+
+    // If the parent directory does not exist, there is nothing to do.
+    if (![fm fileExistsAtPath:container]) {
+        return;
+    }
+
+    NSArray<NSString *> * contents = [fm contentsOfDirectoryAtPath:container error:&error];
+    if (!contents) {
+        bsg_log_err(@"failed to get contents of atomic container: %@", error);
+        return;
+    }
```

Comment:
I would say the correct way to handle this is to ask for the contents of the directory. If it fails, consult the error to see if it failed because the directory doesn't exist. Bail early. Otherwise log the error.